src/nodes.ts

```diff
@@ -129,27 +134,55 @@ export class AttachedServersNode implements INode {
 		return this._attachedServers.length > 0 ? this._attachedServers : [new AttachMongoServerNode()];
 	}
 
-	async attach(connectionString: string): Promise<INode> {
+	async attach(connectionString: string, isInitializing: boolean = false): Promise<INode> {
 		try {
 			const db = await MongoClient.connect(connectionString);
 			const node = new MongoServerNode(connectionString, db.serverConfig);
 			if (this._attachedServers.find(s => s.id === node.id)) {
 				vscode.window.showWarningMessage(`Mongo server '${node.id}' is already attached.`)
 			} else {
 				this._attachedServers.push(node);
+				if (!isInitializing) {
+					await this._azureAccount.credentials.writeSecret(this._connectionStringsId, node.id, connectionString);
```

Comment:
"_serviceName" or "_appName" makes more sense than _connectionStringsId, given the first parameter is called "service".